handlers/officerDeltaHandler_test.go

```diff
@@ -139,17 +171,116 @@ func TestOfficerDeltaHandlerFailsSend(t *testing.T) {
 
 			h := hMocks.NewMockHelper(mockCtrl)
 			svc := sMocks.NewMockKafkaService(mockCtrl)
+			chv := chvMocks.NewMockCHValidator(mockCtrl)
+
+			initEnv()
+			config.CallValidateConfig = func(cfg *config.Config) error {
+				return nil
+			}
 			cfg, _ := config.Get()
 
+			handler := NewOfficerDeltaHandler(svc, h, chv, cfg)
+
+			chv.EXPECT().ValidateRequestAgainstOpenApiSpec(req, handler.cfg.OpenApiSpec).Return(nil, nil)
 			h.EXPECT().GetDataFromRequest(req).Return(requestBody, nil)
-			svc.EXPECT().SendMessage(topic, requestBody).Return(errors.New("error sending message"))
+			svc.EXPECT().SendMessage(handler.cfg.OfficerDeltaTopic, requestBody).Return(errors.New("error sending message"))
+
+			handler.ServeHTTP(resp, req)
+
+			Convey("Then the response should be 500 and an error returned", func() {
+				So(resp.Code, ShouldEqual, http.StatusInternalServerError)
+			})
+
+			destroyEnv()
+		})
+	})
+}
+
+// TestOfficerDeltaHandlerErrorsCallingValidation asserts that the officerDeltaHandler returns an internal error status when validation fails
```

Comment:
If the validation fails then it should be a bad request right ?

Reply:
Comment was updated to clarify that it is indeed when call to validate fails, not a user error.